docs/source/format/CDataInterface/PyCapsuleInterface.rst

```diff
@@ -0,0 +1,238 @@
+  individual libraries.
+
+
+Comparison to DataFrame Interchange Protocol
+--------------------------------------------
+
+:doc:`DataFrame Interchange Protocol <../dataframe.html>` is another protocol
+in Python that allows for the sharing of data between libraries. This protocol
+is complementary to the DataFrame Interchange Protocol. Many of the objects that
+implement this protocol will also implement the DataFrame Interchange Protocol.
+
+This protocol is specific to Arrow-based data structures, while the DataFrame
+Interchange Protocol allows non-Arrow data frames and arrays to be shared as well.
+Because of this, these PyCapsules can support Arrow-specific features such as
+nested columns.
+
+This protocol is also much more minimal than the DataFrame Interchange Protocol.
+It just handles data export, rather than defining accessors for details like
+number of rows or columns.
+
+In summary, if you are implementing this protocol, you should also consider
+implementing the DataFrame Interchange Protocol.
+
+
+PyCapsule Standard
+==================
+
+`PyCapsules`_ allow for a ``name`` to be associated with the capsule, allowing 
+consumers to verify that the capsule contains the expected data. To make sure
+Arrow structs are recognized, the following names should be used:
+
+.. list-table::
+   :widths: 25 25
+   :header-rows: 1
+
+   * - C Interface Type
+     - PyCapsule Name
+   * - ArrowSchema
+     - ``arrowschema``
+   * - ArrowArray
+     - ``arrowarray``
+   * - ArrowArrayStream
+     - ``arrowarraystream``
+
+
+Lifetime Semantics
+------------------
+
+The exported PyCapsules should have a destructor that calls the
+:ref:`release callback <c-data-interface-released>`
+of the Arrow struct, if it is not already null. This prevents a memory leak in
+case the capsule was never passed to another consumer.
+
+If the consumer has been passed to a consumer, the consumer should have moved
+the data and marked the release callback as null, so there isn’t a risk of
+releasing data the consumer is using. Read more in the C Data Interface
+specification.
+
+Because of the lifetime semantics of the underlying Arrow structs, the
+PyCapsules can only be used once.
+
+
+Export Protocol
+===============
+
+The interface is three separate protocols:
+
+* ``ArrowSchemaExportable``, which defines the ``__arrow_c_schema__`` method.
+* ``ArrowArrayExportable``, which defines the ``__arrow_c_array__``. It extends
+  ``ArrowSchemaExportable``, so it must also define ``__arrow_c_schema__``.
+* ``ArrowStreamExportable``, which defines the ``__arrow_c_stream__``. It extends
+  ``ArrowSchemaExportable``, so it must also define ``__arrow_c_schema__``.
+
+The protocols are defined below in terms of ``typing.Protocol``. These may be
+copied into a library for the purposes of static type checking, but this is not
+required to implement the protocol.
+
+.. TODO: should stream have a batch size parameter?
+.. TODO: should we have a allow_copy flag?
```

Comment:
Do folks think we need these parameters?

Reply:
The batch size could potentially also be something that is handled in a protocol one level higher up? (like the Dataset protocol we were discussing, that sparked pushing this one forward)  
I imagine that this interface typically "just" exposes the data as they are, using the batch size as the data already exists in memory. Of course, use cases for an Arrow Stream typically involves data _not_ yet in memory, and there a batch size would be useful (but again, could also be handled by another protocol if you want to be able to control this)